### ISSUE TITLE
Should `success(command)` be renamed to `issucceeded(command)`?

### ISSUE THREAD

**Issue by singularitti** [JuliaLang/julia#50468]
It sounds more compliant with Julia's `is*` naming convention. Also, `process_exited` could be renamed to `is_process_exited`?

https://docs.julialang.org/en/v1/base/base/#Base.success
https://docs.julialang.org/en/v1/base/base/#Base.process_exited

**Comment by Krastanov** [comment]
That is a breaking change, so there is no chance it would happen before a Julia 2.0 is made. A Julia 2.0 would not happen for many years either. And if Julia 2.0 ever happens, I suspect this type of changes would be avoided as they do not significantly increase clarity but create more work for folks that need to port to such a julia 2.0 version.

**Comment by singularitti** [comment]
Would it possible to define an alias like `const success = issucceeded`, etc.? So it is not breaking anymore.

**Comment by Krastanov** [comment]
Yes, but then you probably need a deprecation warning on the old function, which brings up the same issue with this being a change that is of subjective value but it would objectively cause work for package maintainers. (I also like consistency, but here the improvement in consistency is maybe just too small, at the cost of more complicated sounding function names and everything else)

**Comment by PallHaraldsson** [comment]
>then you probably need a deprecation warning on the old function

Do you really? I like consistency, and while I'm not sure how important this function is, I want to discuss this for the general principle.

We could have `const success = issucceeded` for the foreseeable future (even after 2.0) and no "deprecation warning".

What does this const actually do for functions? Does it mean two separate compiled functions, and stored, i.e. bloating the sysimage? Then I don't want that...

We could I guess always do:

```
@noinline success() = issucceded()
```

If we're worried about some old function taking up space, and using that as an example, it would take up minimal code space, and wouldn't even have to be precompiled into the sysimage.

I suppose there's not many technical issues, more of a mental overload to allow synonym functions, but if we are going to consider that, then it seems 1.10 will be the next LTS, so it would be good to get them/this one in, by then, and then only emphasize it in the docs.

**Comment by fingolfin** [comment]
I don't understand what this improves, which practical concern this addresses?

The proposed names also seem to me to be grammatically wrong, `issucceded` sounds really weird (should be `hassucceded`).

**Comment by singularitti** [comment]
> I don't understand what this improves, which practical concern this addresses?

For consistency concern. In Julia, when you want to determine if something is true of false, the function is usually named `is*`, such as `isinteger`, `istaskdone`.

> The proposed names also seem to me to be grammatically wrong, `issucceded` sounds really weird (should be `hassucceded`).

We could name it `hassucceded` if you want. Or `issuccess` just like `issuccess(F::Factorization)` in the `LinearAlgebra` module.

**Comment by IanButterworth** [comment]
I think a key feature of [`success(cmd)`](https://docs.julialang.org/en/v1/base/base/#Base.success) is that it _runs_ the command then reports whether it succeeded. Therefore `is*` or `has*` or even `did*` seems like a bad fit. `run_and_is_successful` would be a reasonable but too verbose option.

Lining suggestions up for a 2.0 change is a good idea, but in this case I'm not sure there's a better suggestion here currently.

**Comment by singularitti** [comment]
@IanButterworth You're right. Maybe [`test_success`](https://github.com/JuliaLang/julia/blob/e4ee485e90961018b7e53ce14b8b99335953e176/base/process.jl#L513-L520) is the one I was looking for. Maybe it could be named `process_succeess(ful)` to be aligned with the other 2 `process_` methods. And we also need 1 `process_failed`.

And the current `succeess` should be named `wait_succeess`, which implies we wait until the process is successful. But, this is bad composition of API. I think a better way is to deprecate this `succeess` method (in 2.0). And export `process_succeess(ful)` & `process_failed`.
